src/extractor/guidance/intersection.cpp

```diff
@@ -9,44 +16,97 @@ namespace guidance
 {
 
 ConnectedRoad::ConnectedRoad(const TurnOperation turn, const bool entry_allowed)
-    : entry_allowed(entry_allowed), turn(turn)
+    : TurnOperation(turn), entry_allowed(entry_allowed)
+{
+}
+
+bool ConnectedRoad::compareByAngle(const ConnectedRoad &other) const { return angle < other.angle; }
+
+void ConnectedRoad::mirror()
+{
+    const constexpr DirectionModifier::Enum mirrored_modifiers[] = {DirectionModifier::UTurn,
+                                                                    DirectionModifier::SharpLeft,
+                                                                    DirectionModifier::Left,
+                                                                    DirectionModifier::SlightLeft,
+                                                                    DirectionModifier::Straight,
+                                                                    DirectionModifier::SlightRight,
+                                                                    DirectionModifier::Right,
+                                                                    DirectionModifier::SharpRight};
+
+    if (angularDeviation(angle, 0) > std::numeric_limits<double>::epsilon())
+    {
+        angle = 360 - angle;
+        instruction.direction_modifier = mirrored_modifiers[instruction.direction_modifier];
```

Comment:
I know this is just moving some code around, but can we assert for out of bounds here?